staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -119,6 +125,25 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 			continue
 		}
 		foundVersion = true
+		if c.openAPIAggregationManager != nil && utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceValidation) {
+			validationSchema, err := getSchemaForVersion(crd, version.Version)
+			if err != nil {
+				return err
+			}
+
+			// We aggregate the schema even if it's nil as it maybe a removal of the schema for this CRD,
+			// and the aggreated OpenAPI spec should reflect this change.
+			crdspec, etag, err := apiextensionsopenapi.CustomResourceDefinitionOpenAPISpec(&crd.Spec, version.Version, validationSchema)
+			if err != nil {
+				return err
+			}
+
+			// Add/update the local API service's spec for the CRD in apiExtensionsServer's
+			// openAPIAggregationManager
+			if err := c.openAPIAggregationManager.AddUpdateLocalAPIServiceSpec(apiServiceName, crdspec, etag); err != nil {
```

Comment:
this seems to be setting the schema computed for a single CRD as the entire schema for the specified `apiServiceName` (which is at the group/version level). Doesn't that mean that if two CRDs were both in the same group/version, only one would have its schema appear in the openapi doc?

Reply:
if that's the case, we need a test to exercise that... two CRDs in the same API group/version get combined properly